### PR TITLE
gobang: unbreak, modernize

### PR DESCRIPTION
**Pull request by pluiedev**
## Description of changes

Bumped the sqlx version used in order to use num-bigint 0.3.3+, which fixes an incompatibility with Rust >1.77 — very similar to the patch used in #309507.

Also changed src to use an SRI hash, reformatted with `nixfmt-rfc-style` and moved to `pkgs/by-name` because why not

Part of ZHF #309482

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by risicle** [comment]
Could you better describe the steps you took to bump sqlx? The problem is this patch file is effectively unreviewable at face value.

**Comment by pluiedev** [comment]
> Could you better describe the steps you took to bump sqlx? The problem is this patch file is effectively unreviewable at face value.

Well, I just updated Cargo.toml to use sqlx 0.5.13 instead of 0.5.6, and ran `cargo check` to rebuild the lock file

File: pkgs/by-name/go/gobang/package.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  darwin,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "gobang";
+  version = "0.1.0-alpha.5";
+
+  src = fetchFromGitHub {
+    owner = "tako8ki";
+    repo = pname;
```
Comment (isabelroses) [inline pkgs/by-name/go/gobang/package.nix:15]: `repo = pname` is an anti pattern, see here https://github.com/NixOS/nixpkgs/issues/277994

File: pkgs/by-name/go/gobang/package.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+  stdenv,
+  darwin,
+}:
+
+rustPlatform.buildRustPackage rec {
```
Comment (isabelroses) [inline pkgs/by-name/go/gobang/package.nix:9]: Consider removing the `rec` for the `let in` syntax see https://nix.dev/guides/best-practices.html

**Comment by pluiedev** [comment]
> > Could you better describe the steps you took to bump sqlx? The problem is this patch file is effectively unreviewable at face value.
> 
> Well, I just updated Cargo.toml to use sqlx 0.5.13 instead of 0.5.6, and ran `cargo check` to rebuild the lock file. I apologize that the diff is this big but apparently the sqlx people change a lot of their deps on minor version bumps 😅 I'll try to find a version that produces a smaller diff

I tried to look for a version of sqlx that produces a smaller diff, but apparently they bumped almost everything in version 0.5.8... Not much we can do then since Cargo doesn't support just manually overriding the version of a transitive dependency like that (the `[patch]` section does not work with crates fulled from crates.io)

**Review by risicle — APPROVED** [review]
Can confirm the `sqlx` bump produces the same result for me.

`nixpkgs-review` happy, macos 12 x86_64 & nixos x86_64

**Pull request merged**